core/src/main/java/org/mule/runtime/core/internal/util/CompositeClassLoader.java

```diff
@@ -45,16 +45,25 @@ public CompositeClassLoader(ClassLoader first, ClassLoader... others) {
   }
 
   @Override
-  public Class<?> loadClass(String name) throws ClassNotFoundException {
+  protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+    Class<?> result = null;
     ClassNotFoundException firstException = null;
     for (ClassLoader classLoader : delegates) {
       try {
-        return classLoader.loadClass(name);
+        result = classLoader.loadClass(name);
+        break;
       } catch (ClassNotFoundException e) {
         firstException = e;
       }
     }
-    throw firstException;
+    if (result == null) {
+      throw firstException;
+    }
+
+    if (resolve) {
+      resolveClass(result);
```

Comment:
why this? shouldn't this be called on the actual classloader instead of the composite?

Reply:
You are right, but we cannot call neither loadClass(s,b) or resolveClass(c) to the delegates as both methods are protected. So, we would need to just ignore the resolve parameter assuming that once the class is "used" it will trigger to load the nested classes (imports, etc) but it will go to the class loader that loaded this class instead of using the CompositeClassLoader (that could ended up finding the class in other class loader).

Reply:
The whole idea of the PR was not change the CompositeClassLoader, instead using it as the TCCL for execution of plugin operations.
This change was only made in order to fix the issue with this class loader when it was introduced in 4.1.5. I will move this change to another PR as there is already a ticket reported for this https://www.mulesoft.org/jira/browse/MULE-18244.